### DISCUSSION TITLE
Storage? (r/w-operations)

### DISCUSSION THREAD

**Discussion by juscghwe** [geerlingguy/pi-nvr#20] [Q&A]
Does the NVR work efficiently with the Pi? An USB outlet somewhat seems deficient to me for the amout of r/w-operations ... And can a regular HDD/SDD handle the amount of r/w's? Or would you recommond specific hardware? Just wanna make sure before going into such an investment.

**Comment by geerlingguy** [comment]
My current office NVR setup is using this Axzez case and a Compute Module 4, which has a full on power supply, hard drives, a SATA breakout board, etc. I made this video about it: https://www.youtube.com/watch?v=7wkVGcdI2vk

If you run just one drive off the Pi using a SATA HAT or an NVMe SSD, many SSDs don't need too much power and are plenty reliable.

I'll soon be migrating my setup to a Pi 5 with an AI + NVMe HAT.

**Reply by juscghwe** [reply]
Thanks for the vid! 🙏 My bad for overlooking it. And thx for the fast insight.

**Comment by msped** [comment]
Hi @geerlingguy, 

Would your migrated setup include the use of something like [PineBoards HAT](https://pineboards.io/products/hatdrive-ai-nvme-2230-2242-gen-2-coral-edge-tpu-for-raspberry-pi-5) with the Coral TPU?

If you have already done a video about the migration I apologise, I'm just being completely blind!

Thank you!

**Reply by geerlingguy** [reply]
Yes, I will be using their AI + NVMe HAT.
